pkg/homedir/homedir.go

```diff
@@ -21,11 +25,14 @@ func GetConfigHome() (string, error) {
 	return filepath.Join(home, ".config"), nil
 }
 
-// GetDataHome returns XDG_DATA_HOME.
+// GetDataHome returns XDG_DATA_HOME for the given UID
 // GetDataHome returns $HOME/.local/share and nil error if XDG_DATA_HOME is not set.
 //
 // See also https://standards.freedesktop.org/basedir-spec/latest/ar01s03.html
-func GetDataHome() (string, error) {
+func GetDataHome(uid int) (string, error) {
```

Comment:
This should no longer change.

Reply:
but shouldn't `GetDataHome` call `GetDataHomeUID`? or do you want `GetDataHomeUID` to be called with `os.Getuid` in this case?

Reply:
No, because we don't need to exec sudo for this. The GetDataHomeUID, is the weird case.

Reply:
ok so these should be separate functions that don't call one another? I thought having one call the other would make sense in some cases but splitting them completely makes sense to me as well.